File: 7. Reverse Integer/solution.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        result = 0
        y = x
        if x < 0:
            y = abs(x)
        while(y):
            remainder = y % 10
            y = int(y / 10)
            result = result * 10 + remainder
            test = result if x > 0 else -result
            if test > pow(2, 31) - 1 or test < -pow(2, 31):
                return 0
        result = result if x > 0 else -result
        return result

File: 7. Reverse Integer/solution.go
func reverse(x int) int {
    var result int = 0
    for {
        if(x == 0) {
            break
        }
        var cur int = x % 10
        x = int(x / 10)
        result = result * 10 + cur
    }
    if(result >2147483647 || result < -2147483648) {
        return 0
    }
    return result
}